### ISSUE TITLE
bindonce repeater

### ISSUE THREAD

**Issue by elemoine** [Pasvaz/bindonce#10]
ng-repeat registers a collection watcher (with $watchCollection), which can be costly, especially when considering ng-repeat based recursive directives (as in http://jsfiddle.net/n8dPm/).

I was wondering if a bindonce type repeater would make sense. That repeater would wait for the collection to be there, but wouldn't shallow-watch the collection.

Not sure it makes sense. Feel free to close this issue if it doesn't. Thanks.

**Comment by Pasvaz** [comment]
Yes it makes sense, I've been thinking a lot about creating bo-repeat in order to get rid of the watcher but not only for that, a repeater uses one watcher ($watchCollection as you said) but what makes it expensive in term of performances is the creation of a child scope for each item, in case of large set can be costly. Now bo-\* fields can live without child scopes because once the item is drawn even if the scope change it is no longer bound with the scope, but if you use any ng-\* field inside the bo-repeater you immediately run into scope issues. 
However we can try to deal with this is it is necessary...

**Comment by akarelas** [comment]
:+1:. I found many cases in my websites where I'd need to use that.

**Comment by adamkdean** [comment]
:+1: It would be one of the most useful cases, especially when grids loop through rows and columns.

**Comment by FGRibreau** [comment]
:+1:

**Comment by mendelk** [comment]
:+1:

**Comment by evilaliv3** [comment]
:+1:

**Comment by MWallenberg** [comment]
I need this in my life. But mostly, I need it in my code... Pasvaz, any time you spend towards making this a reality would be much appreciated :)

**Comment by Pasvaz** [comment]
Hi @MWallenberg, this will be the next step, right now I'm focusing on the rebind feature, when the new binding system will work flawlessy it shouldn't be a problem to implement a repeater.

**Comment by pgayvallet** [comment]
+1, would be awesome to have

**Comment by mparpaillon** [comment]
:+1:

**Comment by philippzentner** [comment]
:+1:

**Comment by schickling** [comment]
:+1: looking forward to this feature

**Comment by fidoboy** [comment]
:+1:

**Comment by steve8708** [comment]
+1

**Comment by akarelas** [comment]
I believe that Angular 1.3 solves this problem, like so:

``` html
<li ng-repeat="item in ::items">{{::item.name}}</li>
```

**Comment by michael-wynn** [comment]
akarelas, I looked at Angular 1.3 documentation and don't see any mention of the syntax you posted.  Where do get that from?

**Comment by akarelas** [comment]
@michael-wynn I don't know where I learned it from (maybe from the Changelog of Angular 1.3) but it works.

**Comment by lorenzhs** [comment]
@michael-wynn @akarelas https://docs.angularjs.org/guide/expression#one-time-binding

**Comment by michael-wynn** [comment]
Thank you!

**Comment by lorenzhs** [comment]
In fact, I replaced bindonce with Angular one-time bindings in Glowing Bear yesterday: https://github.com/glowing-bear/glowing-bear/pull/418

It started with adding `::` to `ng-repeat` statements, but it soon emerged that we could drop bindonce altogether, as we're using angular 1.3 beta in production :bomb: Now we have one less dependency (even if it was only 6.5k minified). Sorry to abandon you ;)
